components/match2/wikis/valorant/big_match.lua

```diff
@@ -244,24 +254,49 @@ function BigMatch:economy(match, opponent1, opponent2)
 			})
 		end
 
-		local chart = mw.ext.Charts.economytimeline({
-			size = {
-				height = 600,
-				width = 800
-			},
-			colors = {'#B12A2A', '#1E7D7D'},
-			groups = {opponent1.name, opponent2.name},
-			data = data,
-			markareas = {
-				{ name = 'Eco', from = 0, to = 6000 },
-				{ name = 'Semi-Eco', from = 6000, to = 14000 },
-				{ name = 'Semi-Buy', from = 14000, to = 20000 },
-				{ name = 'Full-Buy', from = 20000, to = 40000 },
-			},
-		})
+		local function createChart(data, colors)
+			return mw.ext.Charts.economytimeline({
+				size = {
+					height = 600,
+					width = 800
+				},
+				colors = colors,
+				groups = {opponent1.name, opponent2.name},
+				data = data,
+				markareas = {
+					{ name = 'Eco', from = 0, to = 6000 },
+					{ name = 'Semi-Eco', from = 6000, to = 14000 },
+					{ name = 'Semi-Buy', from = 14000, to = 20000 },
+					{ name = 'Full-Buy', from = 20000, to = 40000 },
+				},
+			})
+		end
+
+		local function processHalf(startIndex, endIndex, colors)
+			local data = {}
+			for index = startIndex, endIndex do
+				processRound(index, data)
+			end
+			return #data > 0 and createChart(data, colors) or ''
+		end
+
+		local chart = ''
+		-- First half
+		chart = chart .. processHalf(1,
```

Comment:
Unclear what `1` means in this context. Use a const

Reply:
Added a const